app/importers/file_importers/attendance_importer.rb

```diff
@@ -43,30 +40,39 @@ def import
     log("@skipped_old_rows_count: #{@skipped_old_rows_count}")
     log("@skipped_other_rows_count: #{@skipped_other_rows_count}")
     log('')
-    log("@unchanged_rows_count: #{@unchanged_rows_count}")
-    log("@updated_rows_count: #{@updated_rows_count}")
-    log("@created_rows_count: #{@created_rows_count}")
-    log("@invalid_rows_count: #{@invalid_rows_count}")
-  end
 
-  def remote_file_name
-    LoadDistrictConfig.new.remote_filenames.fetch('FILENAME_FOR_ATTENDANCE_IMPORT', nil)
+    log("Removing records within scope that didn't have CSV rows...")
+    @tardy_record_syncer.delete_unmarked_records!(records_within_scope(Tardy))
+    @absence_record_syncer.delete_unmarked_records!(records_within_scope(Absence))
   end
 
-  def client
-    SftpClient.for_x2
+  private
+  def download_csv
+    client = SftpClient.for_x2
+    remote_file_name = LoadDistrictConfig.new.remote_filenames.fetch('FILENAME_FOR_ATTENDANCE_IMPORT', nil)
+    data_transformer = StreamingCsvTransformer.new(@log)
+    CsvDownloader.new(
+      log: @log,
+      remote_file_name: remote_file_name,
+      client: client,
+      transformer: data_transformer
+    ).get_data
   end
 
-  def data_transformer
-    StreamingCsvTransformer.new(@log)
+  # What existing Insights records should be updated or deleted from running this import?
+  def records_within_scope(record_class)
+    record_class
+      .joins(:student => :school)
+      .where(:schools => {:local_id => @school_local_ids})
+      .where('occurred_at >= ?', @old_threshold)
   end
 
   def school_filter
-    SchoolFilter.new(@school_scope)
+    SchoolFilter.new(@school_local_ids)
   end
 
   def is_old?(row)
-    row[:event_date] < @time_now - 90.days
+    row[:event_date] < @old_threshold
   end
 
   def import_row(row)
```

Comment:
I'm looking at the conditionals here:

```
if !school_filter.include?(row)
  @skipped_from_school_filter = @skipped_from_school_filter + 1
  return
end

if @skip_old_records && is_old?(row)
  @skipped_old_rows_count = @skipped_old_rows_count + 1
  return
end

...
```

It looks like these filters are narrowing down to the exact same set of records as `records_within_scope`:

```
record_class
  .joins(:student => :school)
  .where(:schools => {:local_id => @school_local_ids})
  .where('occurred_at >= ?', @old_threshold)
```

Could having this same logic in two places potentially introduce bugs? 

Since we have to query for `records_within_scope` anyways, what if we query for it at the beginning, use the results to check if each record is within scope in `import_row`, and then pass the same result as an argument to `syncer.delete_unmarked_records!` at the end?

Reply:
Yeah, good question!  These are filtering on different sides.  The "skip" things are filtering the source side (CSV rows), and the `records_within_scope` is filtering on the destination side (Insights records).

So we can't have them literally be the same, and another wrinkle is that the filter on the destination side (Insights records) is applied to two different sets of Insights models.  We could describe the filters as a data structure that is shared by those different filters.  That's kind of what's happening here anyway though by storing `@school_local_ids` and `@old_threshold` but if we factored it out, it would enforce that one of the filters doesn't forget about the other.  Looking at this, I think there might be a bug here where the query doesn't work right without `skip_old_records`, so I'll try this out 👍

Reply:
Ah ok! Thanks for the explanation!